android/src/play/java/com/dooboolab/RNIap/RNIapModule.java

```diff
@@ -676,4 +675,77 @@ private void sendEvent(
         .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
         .emit(eventName, params);
   }
+
+  @ReactMethod
+  public void isSubscribed(final String sku, final Promise promise) {
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchaseHistoryAsync( BillingClient.SkuType.SUBS, (billingResult, list) -> {
+        WritableMap item = Arguments.createMap();
+        for (int i = 0; i < list.size(); i++) {
+          PurchaseHistoryRecord purchase = list.get(i);
+          if (purchase.getSkus().get(0).equalsIgnoreCase(sku)) {
+            promise.resolve(true);
+            return;
+          }
```

Comment:
same here while we are safe because `list.size()`, `getSkus().get(index)` I assume it can also throw

Reply:
you can do a `promise.resolve(false)` inside the catch